src/Reflection/FunctionReflection.php

```diff
@@ -135,7 +135,7 @@ public function getPrototype($format = FunctionReflection::PROTOTYPE_AS_ARRAY)
         $parameters = $this->getParameters();
         foreach ($parameters as $parameter) {
             $prototype['arguments'][$parameter->getName()] = [
-                'type'     => $parameter->getType(),
+                'type'     => $parameter->detectType(),
```

Comment:
@Ocramius — It's used by `Zend\Server\Reflection`, and thus the `Zend\XmlRpc\Server`, `Zend\Json\Server`, and `Zend\Soap\Server` implementations in order to provide the various possible method invocation combinations so that they can:

- provide API usage
- validate provided parameters when processing a request

We moved it *out* of `Zend\Server\Reflection`, IIRC. We could potentially move it back in, but I'd also argue it's a value-add for this component.